hazelcast/src/main/java/com/hazelcast/internal/util/ThreadLocalRandom.java

```diff
@@ -45,14 +39,31 @@
  * <p>This class also provides additional commonly used bounded random
  * generation methods.
  *
- * @since 1.7
+ * Written by Doug Lea with assistance from members of JCP JSR-166
+ * Expert Group and released to the public domain, as explained at
+ * http://creativecommons.org/publicdomain/zero/1.0/
+ *
  * @author Doug Lea
  */
-public class ThreadLocalRandom extends Random {
+public final class ThreadLocalRandom extends Random {
```

Comment:
I had to add a new method, and then I think it's better to integrate the class into our code base. I think we did this with other classes before. Nevertheless, this is just an experiment for now, so this might not be the final solution anyway.